### DISCUSSION TITLE
How to represent between- vs. within-subjects designs?

### DISCUSSION THREAD

**Discussion by emjun** [emitanaka/edibble#30] [Q&A] [answered]
My background: Psychology, computer science

Scenario 1: An educational researcher wants to determine how tutoring affects test scores. So, they assign a group of students into two groups: Tutoring vs. No-Tutoring. The researcher measures exam scores, taken once, and compares the two groups of students. 
I think I would write the edibble program for this like below: 

```
des <- start_design(name = "Tutoring") %>%
        set_units(student = 40) %>%
        set_trts(tutoring = c("No-Tutoring", "Tutoring")) %>%
        allocate_trts(tutoring ~ student) %>%
        randomise_trts()
out <- des %>% 
    set_rcrds(student = c(exam_mark) %>%
```
Scenario 2: Very similar to Scenario 1, but now let's say the educational researcher wants to measure test scores before and after the treatment (Tutoring vs. No-Tutoring). So, in other words, exam score becomes a "repeated measure." 
I think I would start writing the edibble program for this like below: 

```
des <- start_design(name = "Tutoring") %>%
        set_units(student = 40) %>%
        set_trts(tutoring = c("No-Tutoring", "Tutoring")) %>%
        allocate_trts(tutoring ~ student) %>%
        randomise_trts()
out <- des %>% 
    set_rcrds(student = c(exam_mark, when_exam_mark_measured) %>%
```
**Technical question:** **BUT how do I express the repeated measure nature (captured in part by the `when_exam_mark_measured variable`), should it be in the "start_design" or "set_rcrds" clauses?**

**Conceptual question:** What is the rationale behind the answer to the technical question above? 

Additionally, what is the conceptual reason behind not making "Between-subjects" and "Within-subjects" terminology primitives? I understand that they may not be "universal" terms out of human subjects research, but is there also an argument to be said about these ideas around between/within measures as being composed of other primitives in edibble (e.g., units, treatment, records)? If so, what is the rationale? 

Thanks so much!!
Eunice

**Comment by madebyafox** [comment]
Upvoting this question! I'm also a psychological researcher (looking for DoE packages), and I find it difficult to translate between the sets of terminology. Would be happy to collaborate on this if the author is interested!

**Comment by emitanaka** [comment, marked as answer]
I only saw this question now (somehow no notification from GitHub?) so excuse me it's late (feel free to tag me next time).

I've changed the API now so for the first scenario, it will now be:

```r
des1 <- design(name = "Tutoring") %>%
    set_units(student = 40) %>%
    set_trts(tutoring = c("No-Tutoring", "Tutoring")) %>%
    allot_trts(tutoring ~ student) %>%
    assign_trts(order = "random") %>% 
    set_rcrds(exam_mark = student) %>% 
    serve_table()
```

As for the second scenario, you need to set up your units in the _long_ format, so you need to shift how you set up the `unit` structure. So if you want to specify the nature for the first (or before) and second (or after) exam, now you have a factor called `exam_no` that capture this. 

```r
des2 <- design(name = "Tutoring") %>%
  set_units(student = 40, 
            exam_no = c("first", "second"),
            exam_booklet = crossed_by(student, exam_no)) %>%
  set_trts(tutoring = c("No-Tutoring", "Tutoring")) %>%
  allot_trts(tutoring ~ student) %>%
  assign_trts(order = "random") %>% 
  set_rcrds(exam_mark = exam_booklet) %>% 
  serve_table()
 ```

There is a reason I've been avoiding using "between" and "within" subject terminology but I will explain this in more detail in the [edibble book](https://emitanaka.org/edibble-book/) when I get more time at hand.
